cpu/samd21/include/periph_clock_config.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (C) 2017 Daniel Evans <photonthunder@gmail.com>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup         cpu_samd21
+ * @{
+ *
+ * @file
+ * @brief           CPU specific definitions for default clock configuration
+ *
+ * @author          Daniel Evans <photonthunder@gmail.com>
+ */
+
+#ifndef PERIPH_CLOCK_CONFIG_H
+#define PERIPH_CLOCK_CONFIG_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Default Clock configurations
+ */
+    
+/* If RTC or RTT is on Need XOSC32 */
+#if RTC_NUMOF || RTT_NUMOF
+#ifndef CLOCK_XOSC32K
+#define CLOCK_XOSC32K       (32768UL)
+#endif
+#ifndef GEN2_XOSC32
+#define GEN2_XOSC32         (1)
+#endif
+#endif
```

Comment:
I must say I am not so happy with the RTC/RTT dependency here. I would prefer that the cpu clock module merely provides functions to enable/disable the low frequency clock, which then in term can be used by the RTT/RTC. This is how we recently refactored the clock code for the stm32 family...

Reply:
So more precise: IMHO the board config should only define if an external 32khz oscillator is available, and the RTC/RTT modules should then simply request enabling that clock from this module. 

Or of course the XOSC32K clock is enabled during system initialization in case it is configured to be feeding the PLL.